src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java

```diff
@@ -294,17 +294,11 @@ public static GenericStatus computeChunkStatus2(@Nonnull WorkflowRun run,
         }
         ErrorAction err = lastNode.getError();
         if (err != null) {
-            ErrorAction afterError = after.getError();
-            if(afterError != null) {
-                Throwable rootCause = afterError.getError();
-                if (rootCause instanceof FlowInterruptedException) {
-                    return GenericStatus.fromResult(((FlowInterruptedException) rootCause).getResult());
-                } else {
-                    return GenericStatus.FAILURE;
-                }
+            Throwable rootCause = err.getError();
+            if (rootCause instanceof FlowInterruptedException) {
+                return GenericStatus.fromResult(((FlowInterruptedException) rootCause).getResult());
             } else {
-                // If next node lacks an ErrorAction, the error was caught in a catch block
```

Comment:
The comment in this `else` is not clear to me. `catchOutsideFailingStage` demonstrates the test case that I imagined corresponded to this code, but it passes with and without the change. As long as `computeChunkStatus2` is being called for a stage or parallel branch in the ways described in the PR description, I don't think this is possible. (If you pass arbitrary arguments to `computeChunkStatus2` then sure, it could matter, but I think that would be an unintended use of the method)